docs/oras-auth-provider.md

```diff
@@ -0,0 +1,143 @@
+# ORAS Authentication Provider
+
+The ORAS referrer store is responsible for all registry and referrer resolution operations. In order for the ORAS referrer store to access referrer artifacts, authentication credentials must be provided to ORAS. Ratify contains multiple Authentication Providers to support different authentication scenarios within or outside of Kubernetes clusters. The user simply specifies which authentication provider to use in the configuration.
+
+In the configuration file, the user specifies the authentication provider in the `auth-provider` section. The `name` field is **required** to bind ratify to the correct provider implementation. 
+
+## Example config.json
+```
+{
+    "stores": {
+        "version": "1.0.0",
+        "plugins": [
+            {
+                "name": "oras",
+                "localCachePath": "./local_oras_cache",
+                "auth-provider": {
+                    "name": "<auth provider name>",
+                    <other provider specific fields>
+                }
+            }
+        ]
+    },
+    "verifiers": {
+        "version": "1.0.0",
+        "plugins": [
+            {
+                "name":"notaryv2",
+                "artifactTypes" : "application/vnd.cncf.notary.v2.signature",
+                "verificationCerts": [
+                    "<cert folder>"
+                  ]
+            }  
+        ]
+    }
+}
+```
+
+
+NOTE: If the authentication provider cannot resolve credentials for the specified registry host name, it will attempt to use anonymous credentials.
+
+## Supported Providers
+
+### 1. Docker Config
+This is the default authentication provider. If the `auth-provider` section is not specified in the configuration file, ratify will attempt to look for credentials at the default docker configuration path ($HOME/.docker/config.json).
+
+To specify a different docker config file path, the `docker-config` authentication provider can be defined in the configuration with the `configPath` field set. 
+
+```
+"auth-provider": {
+    "name": "docker-config",
+    "configPath": <custom file path string>
+}
+```
+
+### 2. Azure Workload Identity
+Using Workload Federated Identity in an Azure Kubernetes Service cluster, Ratify can pull artifacts from a private Azure Container Registry for verification. For an overview on how workload identity operates in Azure, refer to the [documentation](https://docs.microsoft.com/en-us/azure/active-directory/develop/workload-identity-federation). 
+
+#### User steps to set up Workload Identity with AKS and ACR:
+
+The official steps for setting up Workload Identity on AKS can be found [here](https://azure.github.io/azure-workload-identity/docs/quick-start.html).  
+
+1. Create ACR
+2. Create OIDC enabled AKS cluster by following steps [here](https://docs.microsoft.com/en-us/azure/aks/cluster-configuration#oidc-issuer-preview)
```

Comment:
Add step to add cluster config to kubectl: https://docs.microsoft.com/en-us/azure/aks/tutorial-kubernetes-deploy-cluster?tabs=azure-cli#connect-to-cluster-using-kubectl